File: src/main/java/org/evernet/service/VertexService.java
package org.evernet.service;

import lombok.RequiredArgsConstructor;
import org.evernet.bean.Vertex;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class VertexService {

    private final ConfigService configService;

    public Vertex get() {
        return Vertex.builder()
                .endpoint(configService.getVertexEndpoint())
                .displayName(configService.getVertexDisplayName())
                .description(configService.getVertexDescription())
                .build();
    }
}

File: service/vertex_service.py
from service.config_service import ConfigService


class VertexService:
    def __init__(self, config_service: ConfigService):
        self.config_service = config_service

    def get_info(self):
        return {
            "endpoint": self.config_service.get_vertex_endpoint(),
            "display_name": self.config_service.get_vertex_display_name(),
            "description": self.config_service.get_vertex_description()
        }

    def update_info(self, endpoint, display_name, description):
        self.config_service.set_vertex_endpoint(endpoint)
        self.config_service.set_vertex_display_name(display_name)
        self.config_service.set_vertex_description(description)
